src/test/java/com/iota/iri/network/TransactionRequesterTest.java

```diff
@@ -76,6 +79,48 @@ public void instance() throws Exception {
 
     }
 
+    @Test
+    public void popEldestTransactionToRequest() throws Exception {
+        TransactionRequester txReq = new TransactionRequester(tangle, snapshotProvider, mq);
+        // Add some Txs to the pool and see if the method pops the eldest one
+        Hash eldest = TransactionViewModelTest.getRandomTransactionHash();
+        txReq.requestTransaction(eldest, false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+
+        txReq.popEldestTransactionToRequest();
+        // Check that the transaction is there no more
+        assertEquals(false, txReq.isTransactionRequested(eldest, false));
+    }
+
+    @Test
+    public void transactionRequestedFreshness() throws Exception {
+        // Add some Txs to the pool and see if the method pops the eldest one
+        ArrayList<Hash> eldest = new ArrayList<Hash>(Arrays.asList(
```

Comment:
```suggestion
        List<Hash> eldest = new ArrayList<Hash>(Arrays.asList(
```

Usually in java we like to use the interface for abstraction.
For this unit test it doesn't really matter too much